sdk/go/common/util/cmdutil/profile.go

```diff
@@ -82,12 +82,12 @@ func memoryProfileWriteLoop(prefix string) {
 			fmt.Fprintf(os.Stderr, "could not create memory profile: %s\n", err.Error())
 			return
 		}
-		defer mem.Close()
 
 		runtime.GC() // get up-to-date statistics
 		if err = pprof.Lookup("allocs").WriteTo(mem, 0); err != nil {
 			fmt.Fprintf(os.Stderr, "could not create memory profile: %s\n", err.Error())
```

Comment:
```suggestion
			contract.IgnoreClose(mem)
			fmt.Fprintf(os.Stderr, "could not create memory profile: %s\n", err.Error())
```

I think we should close this in the error case as well (though it probably doesn't matter much in practice).

Reply:
I think this was intended for the other `if`, maybe? I've added it there in any case :crossed_fingers: